cloud/deployment/deployment_variable.go

```diff
@@ -37,42 +41,47 @@ func VariableList(deploymentID, variableKey, ws, envFile, deploymentName string,
 	if currentDeployment.EnvironmentVariables != nil {
 		environmentVariablesObjects = *currentDeployment.EnvironmentVariables
 	}
+	if variableKey != "" {
+		environmentVariablesObjects = slices.DeleteFunc(environmentVariablesObjects, func(v astroplatformcore.DeploymentEnvironmentVariable) bool {
+			return v.Key != variableKey
+		})
+	}
 
 	// open env file
 	if useEnvFile {
-		err = writeVarToFile(environmentVariablesObjects, variableKey, envFile)
+		err = writeVarToFile(environmentVariablesObjects, envFile)
 		if err != nil {
 			fmt.Fprintln(out, errors.Wrap(err, "unable to write environment variables to file"))
 		}
 	}
+	if len(environmentVariablesObjects) == 0 {
+		fmt.Fprintln(out, "\nNo variables found")
+		return nil
+	}
 
-	var nbEnvVarFound int
-	for i := range environmentVariablesObjects {
-		if environmentVariablesObjects[i].Value != nil && !environmentVariablesObjects[i].IsSecret {
-			printValue = *environmentVariablesObjects[i].Value
-		}
-		if environmentVariablesObjects[i].Key == variableKey {
-			nbEnvVarFound++
-			varTab.AddRow([]string{strconv.Itoa(nbEnvVarFound), environmentVariablesObjects[i].Key, printValue, strconv.FormatBool(environmentVariablesObjects[i].IsSecret)}, false)
-			break
-		} else if variableKey == "" {
-			nbEnvVarFound++
-			varTab.AddRow([]string{strconv.Itoa(nbEnvVarFound), environmentVariablesObjects[i].Key, printValue, strconv.FormatBool(environmentVariablesObjects[i].IsSecret)}, false)
+	for i, variable := range environmentVariablesObjects {
+		if variable.Value != nil && !variable.IsSecret {
+			printValue = *variable.Value
```

Comment:
do we not have to reset this to `****`
eg. the first variable is not secret then the next one is, won't do the next `printValue` be the same as the first one?

Reply:
if that is indeed the case, we should add a test case to cover that
if it's not, maybe you can help me fill in the missing gaps im missing

Reply:
<img width="735" alt="Screenshot 2024-02-20 at 6 36 17 PM" src="https://github.com/astronomer/astro-cli/assets/63181127/a734f4d4-97fc-466c-a362-c2c8658c9415">
 seems to be working as intended

Reply:
what if theres another secret var after the third one

Reply:
Agree with Vandy's comment here, can we just declare a temporary variable inside the loop? A unit test case is required here.

Reply:
I didn't actually change this behaviour -- before it was

```go

printValue = *environmentVariablesObjects[i].Value
```

I agree though, while we're here lets fix it

Reply:
Global variables bad!